### PR TITLE
Follow-up to ObjectTemplate code: renames, Multi-doc YAML

### PR DESCRIPTION
**Pull request by Kashomon**
Several followups following some comments from @mikedanese.

- renames TemplateFile to File
- renames TemplateType to Type
- enables multi-doc YAML files

### CODE REVIEW

File: pkg/options/gotmpl/go_template_applier.go [hunk 1]
```diff
@@ -32,6 +33,8 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+var multiDoc = regexp.MustCompile("---(\n|$)")
```
Comment (mikedanese) [inline pkg/options/gotmpl/go_template_applier.go]: Match the beginning of the line as well.

Comment (Kashomon) [reply pkg/options/gotmpl/go_template_applier.go]: `^` will cause the regex match just the beginning of entire string. 

https://play.golang.org/p/NT2W17pdTVt

Comment (mikedanese) [reply pkg/options/gotmpl/go_template_applier.go]: `(\n|^)---(\n|$)` or don't use regex? Is `foo: a---` valid yaml?

Comment (Kashomon) [reply pkg/options/gotmpl/go_template_applier.go]: Ah, right. Done.

**Comment by mikedanese** [comment]
LGTM

**Comment by Kashomon** [comment]
(note: we still use the /lgtm stuff, albeit inconsistently)

**Comment by Kashomon** [comment]
Gonna merge in given the LGTM.

**Pull request merged**